src/test/java/seedu/address/model/client/ClientTest.java

```diff
@@ -145,4 +151,45 @@ public void hashCode_test() {
 
         // TODO: Add tests with modified notes
     }
+
+    @Test
+    public void getClientNotes_modifyUnmodifiableSet_throwsException() {
+        Client client = new ClientBuilder(ALICE).build();
+        Note clientNote = new Note("hell yes");
+        client.addClientNote(clientNote);
+        assertTrue(client.hasClientNote(clientNote));
+        Set<Note> currentNotes = client.getClientNotes();
+        assertThrows(UnsupportedOperationException.class, () -> currentNotes.add(new Note("nice lahh")));
+    }
+
+    @Test
+    public void getClientNotesAsList_modifyUnmodifiableSet_throwsException() {
+        Client client = new ClientBuilder(ALICE).build();
+        client.addClientNote(new Note("hell yes"));
+        List<Note> currentNotes = client.getClientNotesAsList();
+        assertThrows(UnsupportedOperationException.class, () -> currentNotes.add(new Note("nice lahh")));
+    }
+
+    @Test
+    public void deleteClientNote_deleteExistingNote_doesNotThrowExceptionReturnsTrue() {
```

Comment:
Since nothing is returned here, the name should probably be changed. Maybe `deleteClientNote_validNote_noteIsDeleted`?